src/main/lrsql/util/oidc.clj

```diff
@@ -1,19 +1,187 @@
 (ns lrsql.util.oidc
   "OpenID Connect Utilities"
-  (:require [clojure.spec.alpha :as s]
+  (:require [clojure.core.memoize :as mem]
+            [clojure.java.io :as io]
+            [clojure.spec.alpha :as s]
+            [clojure.string :as cs]
+            [clojure.tools.logging :as log]
+            [com.yetanalytics.lrs.auth :as lrs-auth]
             [com.yetanalytics.pedestal-oidc.discovery :as disco]
-            [lrsql.spec.config :as config]))
+            [com.yetanalytics.pedestal-oidc.interceptor :as oidc-i]
+            [com.yetanalytics.pedestal-oidc.jwt :as jwt]
+            [io.pedestal.interceptor :as i]
+            [lrsql.init.authority :as authority]
+            [lrsql.spec.config :as config]
+            [lrsql.spec.oidc :as oidc]
+            [lrsql.util.auth :as auth]
+            [selmer.parser :as selm-parser]
+            [xapi-schema.spec :as xs])
+  (:import [java.io File]))
+
+(defn- select-config
+  [config]
+  (select-keys
+   config
+   [:oidc-issuer
+    :oidc-config
+    :jwks-uri]))
 
 (s/fdef get-configuration
-  :args (s/cat :webserver ::config/webserver)
+  :args (s/cat :config ::config/webserver)
   :ret (s/nilable map?))
 
 (defn get-configuration
   "Given webserver config, return an openid configuration if one is specified
-  via :openid-issuer or :openid-config."
-  [{:keys [openid-issuer
-           openid-config]}]
-  (when-let [config-uri (or openid-config
-                            (and openid-issuer
-                                 (disco/issuer->config-uri openid-issuer)))]
-    (disco/get-openid-config config-uri)))
+  via :oidc-issuer or :oidc-config."
+  [{:keys               [oidc-issuer
+           oidc-config] :as config}]
```

Comment:
Weird formatting?

Reply:
haha clj-fmt edge case

Reply:
fixed in https://github.com/yetanalytics/lrsql/pull/205/commits/50247e22222e58a8d11cbb94e804a599dbc17a0a